core/unit_test/TestNumericTraits.hpp

```diff
@@ -40,7 +40,14 @@ struct extrema {
 
   DEFINE_EXTREMA(float, -FLT_MAX, FLT_MAX);
   DEFINE_EXTREMA(double, -DBL_MAX, DBL_MAX);
+
+// FIXME_NVHPC: with 23.3 using long double in KOKKOS_FUNCTION is hard error
+#if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_COMPILER_NVHPC)
```

Comment:
Presumably you meant
```suggestion
#if !(defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_NVHPC))
```
This broke our Clang+CUDA build